src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -108,54 +108,28 @@ private void HandleClusterDatabaseChanged(string databaseName, long index, strin
                 if (task.IsCanceled || task.IsFaulted)
                     return;
 
+                var database = task.Result;
```

Comment:
This is probably a bad idea. What if we have a long database load time?
You cannot force a wait here.

Reply:
`HandleClusterDatabaseChanged` method is executed asynchronously, so it is not blocking any other operation, but it might eat-up threads.. would `AsyncHelpers.RunSync(() => task)` help here?

Reply:
No, would also hold us the thread.
Instead, change the execute to be async, so we can await there.